src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -229,10 +229,16 @@ let mk_file_match_hook (conf : Scan_CLI.conf) (rules : Rule.rules)
      * get something that Matches_report.pp_text_outputs can operate on
      *)
     let pms : Core_match.t list = match_results.matches in
-    let core_matches : Out.core_match list =
+    (* First process matches with the correct engine_config for nosem filtering *)
+    let processed_matches = 
       pms
-      (* OK, because we don't need the postprocessing to report the matches. *)
       |> List_.map Core_result.mk_processed_match
+    in
+    let processed_matches, _errors =
+      Nosemgrep.produce_ignored ~config:conf.core_runner_conf.engine_config processed_matches
+    in
+    let core_matches : Out.core_match list =
+      processed_matches
```

Comment:
I wonder why that is needed here, given that the standard `nosem` worked -- are we sure this is the right place?

Reply:
In particular we already have this kind of call in `src/core_scan/Pre_post_core_scan.ml`, module `Nosemgrep_processor`.

Reply:
Ok, not convinced about this.
We have this in 2 places, it should be in the post-processor module.

Investigating if this a config issue, ie, is the right config available to the post-processor?

Reply:
In fact since with `--experimental` this already works, this change here is not necessary!

The problem is when `--experimental` is not passed!